src/test/functions.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+require '../functions.php';
+
+use PHPUnit\Framework\TestCase; // magic to get PHPUnit testing to work (lookup)
+
+class Functions extends TestCase
+{
+	public function test_Success_Display_Cosmic_Event() {
```

Comment:
need to also provide a failure test with an array that doesn't have the appropriate format expected by the function (this will probably require you to change the original function as well)

Reply:
db_functions separated and unit tests complete for every testable function